src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -193,7 +195,19 @@ public void ValidateAppInsightsTelemetryConnectionString(RuntimeConfig runtimeCo
     /// </summary>
     public async Task<JsonSchemaValidationResult> ValidateConfigSchema(RuntimeConfig runtimeConfig, string configFilePath, ILoggerFactory loggerFactory)
     {
-        string jsonData = _fileSystem.File.ReadAllText(configFilePath);
+        string? jsonData = _fileSystem.File.ReadAllText(configFilePath);
+
+        // The config file may contain some environment variables that need to be replaced before validation.
+        try
+        {
+            jsonData = Utf8JsonReaderExtensions.ReplaceEnvVarsInJson(jsonData);
```

Comment:
We already have the `runtimeConfig` deserialized from the `configFilePath`. 

Wouldnt that mean we already had the jsonData with the environment variables replaced? 
Or does the runtimeConfig object doesnt have the environment variable values? 
Instead of reading from the file path, cant we get the string from which the `runtimeConfig` object was formed - to do the schema validation on?

Reply:
no we cant. schemaValidator require a json string. i tried converting the deserialized runtimeConfig to string and then passing it to the schema validator but it was not forming a valid json string.

Reply:
can you provide an example? invalid json string because of extra escape characters? is 'invalid json string' indicative of a different bug?

Reply:
no by invalid i mean.
if i deserialize the below json 
```
{
"database-type": "mssql"
}
```

and then deserialize it
```
{
"database-type" : {
"DatabaseType": 2
}
}
```

Reply:
that seems like another bug

Reply:
it's just how enums are serialized i believe, don't think it's a bug

Reply:
> it's just how enums are serialized i believe, don't think it's a bug

I think we would need to have custom (de)serialization that handles the case of these enums so that they go back to the intended format. Deserialization/Serialization ought to be reversable so this does seem like a bug that just hasnt arisen yet.

Reply:
using `string jsonData = runtimeConfig.ToJson()` instead of all the custom parsing code and then provided `jsonData` to:
`await jsonConfigSchemaValidator.ValidateJsonConfigWithSchemaAsync(jsonSchema, jsonData!);`

works great and doesn't present any of the issues mentioned earlier in this thread. Am I missing something?

Reply:
do the tests added in this PR pass with your proposed changes? If yes, we may not need the custom parsing code mentioned here.